app/helpers/searchHelpers.php

```diff
@@ -432,3 +432,166 @@ function caGetChangeLogForElasticSearch($po_db, $pn_table_num, $pn_row_id) {
+			// Convert CA field type to query builder type and operators.
+			switch ($vo_field_info['FIELD_TYPE']) {
+				case FT_NUMBER:
+					$va_result['type'] = 'integer';
+					break;
+				case FT_DATE:
+				case FT_DATERANGE:
+				case FT_HISTORIC_DATE:
+				case FT_HISTORIC_DATERANGE:
+					$va_result['type'] = 'date';
+					break;
+				case FT_TIME:
+				case FT_TIMECODE:
+				case FT_TIMESTAMP:
+				case FT_TIMERANGE:
+					$va_result['type'] = 'time';
+					break;
+				case FT_DATETIME:
+				case FT_HISTORIC_DATETIME:
+					$va_result['type'] = 'datetime';
+					break;
+				default:
+					$va_result['type'] = 'string';
+			}
+		} elseif (in_array($vs_name_no_table, $va_element_codes)) {
+			$t_element = ca_metadata_elements::getInstance($vs_name_no_table);
+			if ($t_element) {
+				// Get the list code and display type for further processing below.
+				$vs_list_code = $t_element->get('list_id');
+				$vn_display_type = $vs_list_code ? DT_SELECT : DT_FIELD;
+				// Convert CA attribute datatype to query builder type and operators.
+				switch ($t_element->get('datatype')) {
+					case __CA_ATTRIBUTE_VALUE_CURRENCY__:
+					case __CA_ATTRIBUTE_VALUE_LENGTH__:
+					case __CA_ATTRIBUTE_VALUE_NUMERIC__:
+					case __CA_ATTRIBUTE_VALUE_WEIGHT__:
+						$va_result['type'] = 'double';
+						break;
+					case __CA_ATTRIBUTE_VALUE_INTEGER__:
+						$va_result['type'] = 'integer';
+						break;
+					case __CA_ATTRIBUTE_VALUE_DATERANGE__:
+						$va_result['type'] = 'date';
+						break;
+					case __CA_ATTRIBUTE_VALUE_TIMECODE__:
+						$va_result['type'] = 'time';
+						break;
+					default:
+						$va_result['type'] = 'string';
+						break;
+				}
+			}
+		}
+		// Use the relevant input field type and operators based on type.
+		$va_result['operators'] = $vo_operators_by_type[$va_result['type']];
+		// Process list types and use a text field for non-list types.
+		if (in_array($vn_display_type, array( DT_SELECT, DT_LIST, DT_LIST_MULTIPLE, DT_CHECKBOXES, DT_RADIO_BUTTONS ))) {
+			if (!$va_select_options) {
+				$va_select_options = array();
+				$t_list = new ca_lists();
+				$va_items = $t_list->getItemsForList($vs_list_code);
+				if (is_array($va_items)) {
+					foreach ($va_items as $va_item) {
+						foreach ($va_item as $va_item_details) {
+							$va_select_options[$va_item_details['idno']] = $va_item_details['name_singular'];
+						}
+					}
+				}
+			}
+			$va_result['input'] = 'select';
+			$va_result['values'] = $va_select_options;
+			$va_result['operators'] = $vo_operators_by_type['select'];
+		} else {
+			$va_result['input'] = 'text';
+		}
+		// Set up option groups.
+		if (in_array($vs_name, $va_priority)) {
+			// Bundle is given priority
+			$va_result['optgroup'] = 'Frequently used fields';
+		} else {
+			$vn_split = strpos($vs_name, '.');
+			if ($vn_split === false || substr($vs_name, 0, $vn_split) === $vs_table) {
+				$va_result['optgroup'] = 'Fields in ' . $t_subject->getProperty('NAME_PLURAL');
```

Comment:
Will fix this too - `_t('Fields in %1', $t_subject->getProperty('NAME_PLURAL');`  (not sure of translation syntax will check on `%1`)